src/statemachine/statemachine_coroutine.js

```diff
@@ -0,0 +1,225 @@
+pc.extend(pc, (function () {
```

Comment:
In framework_application.js in the constructor. I'd do something like this:

```
this._statemachineManager = new pc.StateMachineManager();
```

Then in `update()`

You could just call

```
this._stateMachineManager.update(dt);
```